### PR TITLE
Remove IncrementalTaskInputs from the docs

### PR DESCRIPTION
**Pull request by asodja**
This removed `IncrementalTaskInputs` from docs for 8.0. `IncrementalTaskInputs` are now mentioned just in the upgrade guide.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/extending-gradle/custom_tasks.adoc [hunk 1]
```diff
@@ -139,17 +139,6 @@ This can be particularly useful for a transformer task that converts input files
...
 If you'd like to optimize your build so that only out-of-date input files are processed, you can do so with an _incremental task_.
 
-[NOTE]
-====
-There is the link:{javadocPath}/org/gradle/api/tasks/incremental/IncrementalTaskInputs.html[IncrementalTaskInputs] API, which is available in Gradle versions before 5.4.
-When using link:{javadocPath}/org/gradle/api/tasks/incremental/IncrementalTaskInputs.html[IncrementalTaskInputs], it is only possible to query for all file changes of the task inputs.
-It is not possible to query for changes of individual input file properties.
-Moreover, the old API does not distinguish between incremental and non-incremental task inputs, so the task itself needs to determine where the changes originated from.
-Therefore, the usage of this API is deprecated, and it will be removed eventually.
-The new link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges] API, which is documented here, replaces the old API and addresses its shortcomings.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/userguide/extending-gradle/custom_tasks.adoc:149]: Should we add this "new link" to the previous sentence?

File: subprojects/docs/src/docs/userguide/extending-gradle/custom_tasks.adoc [hunk 2]
```diff
@@ -139,17 +139,6 @@ This can be particularly useful for a transformer task that converts input files
 
 If you'd like to optimize your build so that only out-of-date input files are processed, you can do so with an _incremental task_.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/userguide/extending-gradle/custom_tasks.adoc:140]:
```suggestion
You can optimize your build to only process out-of-date input files with an _incremental task_. To learn more, refer to the link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges] API documentation.
```

**Review by nathan-contino — APPROVED** [review]
Looks good, but I think we should include the link to the `InputChanges` API still. Added a suggestion for that.

File: subprojects/docs/src/docs/userguide/extending-gradle/custom_tasks.adoc (hunk 2 above)
Comment (asodja) [reply subprojects/docs/src/docs/userguide/extending-gradle/custom_tasks.adoc:140]: Note that after that it's described how you can use InputChanges and API link is in second sentence of the next paragraph. Check: https://docs.gradle.org/current/userguide/custom_tasks.html#incremental_tasks. 

```
For a task to process inputs incrementally, that task must contain an _incremental task action_.
This is a task action method that has a single link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges] parameter.
```

Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/extending-gradle/custom_tasks.adoc:140]: Thanks for pointing that out. Feel free to disregard this then.

**Comment by lptr** [comment]
We should mention the removal in `upgrading_version_7.adoc`, too, right?

**Review by lptr — COMMENTED** [review]
We should mention the removal in `upgrading_version_7.adoc`, too, right?

Apart from the update docs LGTM.

**Comment by asodja** [comment]
Preview:
Screenshot 2022-08-26 at 13 14 46

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 3]
```diff
@@ -59,8 +59,22 @@ For more information, see <<tar_tree_no_backing_file>>.
...
+==== Legacy IncrementalTaskInputs API
+
+The legacy `IncrementalTaskInputs` API has been removed. For more information, see <<incremental_task_inputs_deprecation>>. This change also affects Kotlin Gradle Plugin and Android Gradle Plugin. With Gradle 8.0 you should use Kotlin Gradle Plugin 1.6.10 or later and Android Gradle Plugin 7.3.0 with `android.experimental.legacyTransform.forceNonIncremental=true` property or later.
+
 === Potential breaking changes
 
+==== Minimal supported Kotlin Gradle Plugin version changed
+With Gradle 7.x Kotlin Gradle Plugin 1.3.72 through 1.6.21 were tested and supported.
+With Gradle 8.x minimal Kotlin Gradle Plugin tested and supported version is 1.6.10.
+It is still possible to compile to lower Kotlin language version, by modifying language version and api version setting for the Kotlin compile task.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:71]:
```suggestion
Gradle 7.x supports Kotlin Gradle Plugin 1.3.72 and above. Kotlin Gradle Plugin versions above 1.6.21 are not tested with Gradle 7.x.
Gradle 8.x supports Kotlin Gradle Plugin 1.6.10 and above.
You can use a lower Kotlin language version by modifying the language version and api version setting in the Kotlin `compile` task.
```
Suggestion: Use active voice and parallel wording to make these changes easier to understand.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 4]
```diff
@@ -59,8 +59,22 @@ For more information, see <<tar_tree_no_backing_file>>.
...
+
+==== Minimal supported Android Gradle Plugin version changed
+With Gradle 7.x Android Gradle Plugin 4.1 through 7.3 were tested and supported.
+With Gradle 8.x minimal Android Gradle Plugin tested and supported version is 7.3 with `android.experimental.legacyTransform.forceNonIncremental=true` property.
+Android Gradle Plugin 8 and later don't require additional property to work with Gradle 8.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:76]:
```suggestion
Gradle 7.x supports Android Gradle Plugin (AGP) 4.1 and above. AGP versions above 7.3 are not tested with Gradle 7.x.
Gradle 8.x supports AGP 8 and above.
Gradle 8.x supports AGP 7.3 and above if you configure the following property:
`android.experimental.legacyTransform.forceNonIncremental=true`
```

Suggestion: Active voice and parallel wording, as above. Reorder version support to mention the default supported version first, and the strings-attached support second.

**Review by nathan-contino — APPROVED** [review]
Docs LGTM -- two suggestions that I hope you consider.

**Comment by asodja** [comment]
@bot-gradle test and merge

**Pull request merged**